spec/features/groups/group_freezing_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.feature 'Group locking' do
+  context 'suite size still available, full group' do
+    it 'can be initiated by a leader' do
+      group = FactoryGirl.create(:full_group, size: 2)
+      log_in group.leader
+      visit draw_group_path(group.draw, group)
+      click_on 'Finalize Group'
+      expect(group.locked_members).to eq([group.leader])
+      expect(group.reload.finalizing?).to be_truthy
+    end
+    it 'can be confirmed by a member' do
+      group = FactoryGirl.create(:full_group, size: 2)
+      GroupFinalizer.finalize(group: group)
+      member = group.members.last
+      log_in member
+      visit draw_group_path(group.draw, group)
+      click_on 'Finalize Membership'
+      expect(group.locked_members).to eq(group.members)
+      expect(group.locked?).to be_truthy
+    end
+  end
+  context 'suite size not available, as group leader' do 
+    it 'cannot be initiated' do
```

Comment:
I think I decided against testing this failure case, but let me know if you want this text